metrics.go

```diff
@@ -0,0 +1,207 @@
+package metrics
+
+import (
+	"encoding/json"
+	"os"
+
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/alexcesaro/statsd"
+	"github.com/transcovo/go-chpr-logger"
+)
+
+var defaultClientBuilder *clientBuilder
+
+func defaultClientErrorHandler(err error) {
+	// Do we choose to make the metrics depend on the logger ?
+	// That sounds like a bad idea to me but this would be useful to have logs about the connection to the statsd server
+	// fmt.Println("Error caught in metrics", error)
+	logger.WithField("error", err).Error("[Metrics] Error caught in metrics")
+}
+
+func init() {
+	defaultClientBuilder = &clientBuilder{
+		errorLogger: defaultClientErrorHandler,
+	}
+}
+
+/*
+Struct responsible for keeping all the statsd clients
+*/
+type metricsSender struct {
+	Clients []*statsd.Client
+}
+
+var metrics *metricsSender = &metricsSender{}
+
+/*
+Struct target from METRICS_DESTINATION env var
+*/
+type metricsDestination struct {
+	CacheDNS string `json:"METRICS_CACHE_DNS"` // Do we need to keep this since this the go client does not handle it
+	Host     string `json:"METRICS_HOST"`
+	Port     string `json:"METRICS_PORT"`
+	Prefix   string `json:"METRICS_PREFIX"`
+}
+
+/*
+Array of metricDestination (retrieve each metric destination from METRIC_DESTINATIONS)
+*/
+type metricsDestinations []*metricsDestination
+
+/*
+The config to pass to a client builder
+*/
+type clientConfig struct {
+	Host, Prefix string
+	ErrorHandler func(error)
+}
+
+/*
+Return a client for a metric destination struct
+*/
+func clientConfigFromDestination(destination *metricsDestination) *clientConfig {
+	return &clientConfig{
+		Host:   destination.Host + ":" + destination.Port,
+		Prefix: destination.Prefix,
+	}
+}
+
+/*
+Struct responsible for building clients
+*/
+type clientBuilder struct {
+	errorLogger func(error)
+}
+
+/*
+Return a statsd client from our custom clientConfig
+statsd.new will resolve the given host and panics if it fails
+The statsd lib we use sends a message upon connection
+if there is nothing listening it may crash
+TODO : check that this code is removed https://github.com/alexcesaro/statsd/blob/master/conn.go#L50
```

Comment:
There was quite a debate about it, Farzad wanted me to fork the library but I'm not really comfortable with the idea of diverging from potentially bug fixes